tests/measurements/test_sample.py

```diff
@@ -13,23 +13,54 @@
 # limitations under the License.
 """Unit tests for the sample module"""
 
+import sys
+
 import numpy as np
 import pytest
 
 import pennylane as qml
 from pennylane.measurements import Sample
+from pennylane.operation import EigvalsUndefinedError, Operator
 
 
 class TestSample:
     """Tests for the sample function"""
 
-    def test_sample_dimension(self):
+    # TODO: Remove this when new CustomMP are the default
+    def teardown_method(self):
+        """Method called at the end of every test. It loops over all the calls to
+        QubitDevice.sample and compares its output with the new _Sample.process method."""
+        if not getattr(self, "spy", False):
+            return
+        if sys.version_info[1] <= 7:
+            return  # skip tests for python@3.7 because call_args.kwargs is a tuple instead of a dict
+
+        assert len(self.spy.call_args_list) > 0  # make sure method is mocked properly
+
+        samples = self.dev._samples
+        for call_args in self.spy.call_args_list:
+            meas = call_args.args[1]
+            shot_range, bin_size = (call_args.kwargs["shot_range"], call_args.kwargs["bin_size"])
+            if isinstance(meas, Operator):
+                meas = qml.sample(op=meas)
+            assert qml.math.allequal(
+                self.dev.sample(call_args.args[1], **call_args.kwargs),
+                meas.process_samples(
+                    samples=samples,
+                    wire_order=self.dev.wires,
+                    shot_range=shot_range,
+                    bin_size=bin_size,
+                ),
+            )
```

Comment:
I'm not sure if we should have testing code in the teardown method. It's only meant for cleanup after the test runs.

Also the fact that you're setting `self.dev` and `self.spy` just for this is kind of odd. Personally I'd prefer duplicating this code in all of the tests.

Reply:
Yeah you're probably right. Changed 👍